### ISSUE TITLE
Every save, even with a minor change, changes every text layers.

### ISSUE THREAD

**Issue by ChrisZannoni** [kactus-io/kactus#60]
### Description

When I open a sketch file from Kactus then make a small text change and save the sketch file it says the every json file has change. Using text analysis the only change this "_archive" key value pair. Please see attached screen shot 
![screen shot 2017-09-21 at 1 41 18 pm](https://user-images.githubusercontent.com/10788917/30710142-9834d9c2-9ed2-11e7-88c8-377a1a88c241.png)

### Version

**Kactus version:** 0.2.6

**OS version:** macOS Sierra Version 10.12.6

### Steps to Reproduce

1. Open up kactus
1. Open Sketch file with katus
1. Change existing text in an artboard by 1 character.
1. Hit save in sketch
1. Go to Kactus and observe that there all json files have changed. It basically thinks everything was added then deleted.

**Expected behavior:** Only text that was changed shows as changed. And not 507 files.

**Actual behavior:** It show that 507 files have changed.

**Reproduces how often:** Everytime

### Logs

2017-09-21T14:44:03.179Z - error: [ui] Error performing periodic fetch
TypeError: Cannot read property 'do_objectID' of undefined
    at pages.find.p (/private/var/folders/lw/w6fz3m5d7x53qvljf8nfkxbr0000gn/T/AppTranslocation/C4C92867-0A7E-405F-A281-F2BD483F46FF/d/Kactus.app/Contents/Resources/app/node_modules/kactus-cli/lib/parse/parseDocumentJSON.js:30:42)
    at Array.find (native)
    at data.pages.reduce (/private/var/folders/lw/w6fz3m5d7x53qvljf8nfkxbr0000gn/T/AppTranslocation/C4C92867-0A7E-405F-A281-F2BD483F46FF/d/Kactus.app/Contents/Resources/app/node_modules/kactus-cli/lib/parse/parseDocumentJSON.js:30:31)
    at Array.reduce (native)
    at zip.file.async.then (/private/var/folders/lw/w6fz3m5d7x53qvljf8nfkxbr0000gn/T/AppTranslocation/C4C92867-0A7E-405F-A281-F2BD483F46FF/d/Kactus.app/Contents/Resources/app/node_modules/kactus-cli/lib/parse/parseDocumentJSON.js:29:37)
2017-09-21T14:47:25.178Z - info: [ui] Executing getCommits: git log HEAD --date=raw --max-count=100 --pretty=%H%x1F%s%x1F%b%x1F%an <%ae> %ad%x1F%P -z --no-color --not --remotes (took 21.813s)
2017-09-21T14:47:25.180Z - info: [ui] Executing getAuthorIdentity: git var GIT_AUTHOR_IDENT (took 21.808s)
2017-09-21T14:47:53.557Z - info: [ui] Executing updateIndex: git update-index --add --remove --replace -z --stdin (took 2.877s)
2017-09-21T14:47:54.822Z - info: [ui] Executing createCommit: git commit -F - (took 1.265s)
2017-09-21T14:48:19.268Z - info: [ui] Executing push: git -c credential.helper= push origin dev --set-upstream --progress (took 20.148s)
2017-09-21T14:53:43.539Z - info: [ui] Executing getStatus: git status --untracked-files=all --branch --porcelain=2 -z (took 1.314s)
2017-09-21T15:16:00.557Z - info: [ui] Executing exportCommit: git archive 0403e2ca1b6bdf8bd12f0f6e44dd7f8525fb92f4 -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/0403e2ca1b6bdf8bd12f0f6e44dd7f8525fb92f4.tar (took 1.870s)
2017-09-21T15:16:04.490Z - info: [ui] Executing exportCommit: git archive 0403e2ca1b6bdf8bd12f0f6e44dd7f8525fb92f4 -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/0403e2ca1b6bdf8bd12f0f6e44dd7f8525fb92f4.tar (took 1.132s)
2017-09-21T15:16:17.876Z - info: [ui] Executing exportCommit: git archive 0403e2ca1b6bdf8bd12f0f6e44dd7f8525fb92f4 -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/0403e2ca1b6bdf8bd12f0f6e44dd7f8525fb92f4.tar (took 1.298s)
2017-09-21T15:16:20.974Z - info: [ui] Executing exportCommit: git archive 0403e2ca1b6bdf8bd12f0f6e44dd7f8525fb92f4 -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/0403e2ca1b6bdf8bd12f0f6e44dd7f8525fb92f4.tar (took 1.460s)
2017-09-21T15:16:22.363Z - info: [ui] Executing exportCommit: git archive 0403e2ca1b6bdf8bd12f0f6e44dd7f8525fb92f4 -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/0403e2ca1b6bdf8bd12f0f6e44dd7f8525fb92f4.tar (took 1.667s)
2017-09-21T15:16:24.320Z - info: [ui] Executing exportCommit: git archive 0403e2ca1b6bdf8bd12f0f6e44dd7f8525fb92f4 -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/0403e2ca1b6bdf8bd12f0f6e44dd7f8525fb92f4.tar (took 1.298s)
2017-09-21T15:16:32.885Z - info: [ui] Executing exportCommit: git archive 0403e2ca1b6bdf8bd12f0f6e44dd7f8525fb92f4 -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/0403e2ca1b6bdf8bd12f0f6e44dd7f8525fb92f4.tar (took 1.075s)
2017-09-21T15:16:40.588Z - info: [ui] Executing exportCommit: git archive 0403e2ca1b6bdf8bd12f0f6e44dd7f8525fb92f4 -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/0403e2ca1b6bdf8bd12f0f6e44dd7f8525fb92f4.tar (took 1.110s)
2017-09-21T15:17:02.271Z - info: [ui] Executing updateIndex: git update-index --add --remove --replace -z --stdin (took 1.347s)
2017-09-21T15:17:07.698Z - info: [ui] Executing exportCommit: git archive 64e0c24a53bd4b2b31dfb9688be28a5641c4bd7e -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/64e0c24a53bd4b2b31dfb9688be28a5641c4bd7e.tar (took 1.135s)
2017-09-21T15:17:08.658Z - info: [ui] Executing exportCommit: git archive 64e0c24a53bd4b2b31dfb9688be28a5641c4bd7e -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/64e0c24a53bd4b2b31dfb9688be28a5641c4bd7e.tar (took 1.333s)
2017-09-21T15:17:15.487Z - info: [ui] Executing push: git -c credential.helper= push origin testing_chris --set-upstream --progress (took 9.700s)
2017-09-21T15:17:21.003Z - info: [ui] Executing exportCommit: git archive 64e0c24a53bd4b2b31dfb9688be28a5641c4bd7e -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/64e0c24a53bd4b2b31dfb9688be28a5641c4bd7e.tar (took 1.113s)
2017-09-21T15:18:59.510Z - info: [ui] Executing exportCommit: git archive 64e0c24a53bd4b2b31dfb9688be28a5641c4bd7e -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/64e0c24a53bd4b2b31dfb9688be28a5641c4bd7e.tar (took 1.160s)
2017-09-21T15:19:09.122Z - info: [ui] Executing exportCommit: git archive 64e0c24a53bd4b2b31dfb9688be28a5641c4bd7e -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/64e0c24a53bd4b2b31dfb9688be28a5641c4bd7e.tar (took 1.306s)
2017-09-21T15:19:41.138Z - info: [ui] Executing exportCommit: git archive ff17f3b04931a2536b682bb2dc16616d8f1a398c -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/ff17f3b04931a2536b682bb2dc16616d8f1a398c.tar (took 1.140s)
2017-09-21T15:19:46.405Z - info: [ui] Executing exportCommit: git archive ff17f3b04931a2536b682bb2dc16616d8f1a398c -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/ff17f3b04931a2536b682bb2dc16616d8f1a398c.tar (took 1.134s)
2017-09-21T15:19:57.301Z - info: [ui] Executing exportCommit: git archive ff17f3b04931a2536b682bb2dc16616d8f1a398c -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/ff17f3b04931a2536b682bb2dc16616d8f1a398c.tar (took 1.231s)
2017-09-21T15:21:15.359Z - info: [ui] Executing exportCommit: git archive ff17f3b04931a2536b682bb2dc16616d8f1a398c -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/ff17f3b04931a2536b682bb2dc16616d8f1a398c.tar (took 1.135s)
2017-09-21T15:21:21.370Z - info: [ui] Executing pull: git -c credential.helper= pull --no-rebase --progress origin (took 3.430s)
2017-09-21T16:50:46.493Z - info: [ui] Executing exportCommit: git archive ff17f3b04931a2536b682bb2dc16616d8f1a398c -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/ff17f3b04931a2536b682bb2dc16616d8f1a398c.tar (took 1.183s)
2017-09-21T16:50:49.223Z - info: [ui] Executing exportCommit: git archive ff17f3b04931a2536b682bb2dc16616d8f1a398c -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/ff17f3b04931a2536b682bb2dc16616d8f1a398c.tar (took 1.314s)
2017-09-21T16:50:55.567Z - info: [ui] Executing exportCommit: git archive ff17f3b04931a2536b682bb2dc16616d8f1a398c -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/ff17f3b04931a2536b682bb2dc16616d8f1a398c.tar (took 1.282s)
2017-09-21T16:51:07.446Z - info: [ui] Executing exportCommit: git archive ff17f3b04931a2536b682bb2dc16616d8f1a398c -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/ff17f3b04931a2536b682bb2dc16616d8f1a398c.tar (took 1.193s)
2017-09-21T16:51:37.722Z - info: [ui] Executing exportCommit: git archive ff17f3b04931a2536b682bb2dc16616d8f1a398c -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/ff17f3b04931a2536b682bb2dc16616d8f1a398c.tar (took 1.186s)
2017-09-21T16:51:46.385Z - info: [ui] Executing exportCommit: git archive ff17f3b04931a2536b682bb2dc16616d8f1a398c -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/ff17f3b04931a2536b682bb2dc16616d8f1a398c.tar (took 1.188s)
2017-09-21T16:51:51.257Z - info: [ui] Executing exportCommit: git archive ff17f3b04931a2536b682bb2dc16616d8f1a398c -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/ff17f3b04931a2536b682bb2dc16616d8f1a398c.tar (took 1.347s)
2017-09-21T16:52:00.401Z - info: [ui] Executing exportCommit: git archive 64e0c24a53bd4b2b31dfb9688be28a5641c4bd7e -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/64e0c24a53bd4b2b31dfb9688be28a5641c4bd7e.tar (took 1.132s)
2017-09-21T16:52:07.352Z - info: [ui] Executing exportCommit: git archive 0403e2ca1b6bdf8bd12f0f6e44dd7f8525fb92f4 -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/0403e2ca1b6bdf8bd12f0f6e44dd7f8525fb92f4.tar (took 1.511s)
2017-09-21T16:52:28.390Z - info: [ui] Executing getStatus: git status --untracked-files=all --branch --porcelain=2 -z (took 2.425s)
2017-09-21T16:52:28.443Z - info: [ui] Executing getStatus: git status --untracked-files=all --branch --porcelain=2 -z (took 1.310s)
2017-09-21T16:52:31.257Z - info: [ui] Executing exportCommit: git archive ff17f3b04931a2536b682bb2dc16616d8f1a398c -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/ff17f3b04931a2536b682bb2dc16616d8f1a398c.tar (took 1.270s)
2017-09-21T16:52:31.263Z - info: [ui] Executing exportCommit: git archive ff17f3b04931a2536b682bb2dc16616d8f1a398c -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/ff17f3b04931a2536b682bb2dc16616d8f1a398c.tar (took 1.357s)
2017-09-21T16:52:31.270Z - info: [ui] Executing exportCommit: git archive ff17f3b04931a2536b682bb2dc16616d8f1a398c -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/ff17f3b04931a2536b682bb2dc16616d8f1a398c.tar (took 1.313s)
2017-09-21T16:52:31.634Z - info: [ui] Executing exportCommit: git archive ff17f3b04931a2536b682bb2dc16616d8f1a398c -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/ff17f3b04931a2536b682bb2dc16616d8f1a398c.tar (took 1.747s)
2017-09-21T16:54:12.330Z - info: [ui] Executing exportCommit: git archive ff17f3b04931a2536b682bb2dc16616d8f1a398c -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/ff17f3b04931a2536b682bb2dc16616d8f1a398c.tar (took 1.168s)
2017-09-21T16:54:23.892Z - info: [ui] Executing exportCommit: git archive ff17f3b04931a2536b682bb2dc16616d8f1a398c -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/ff17f3b04931a2536b682bb2dc16616d8f1a398c.tar (took 1.260s)
2017-09-21T16:55:44.068Z - error: [main] Error: write EPIPE
    at exports._errnoException (util.js:1022:11)
    at WriteWrap.afterWrite (net.js:804:14)
2017-09-21T16:55:44.074Z - error: [ui] `git checkout-index -f -u -q --stdin -z` exited with an unexpected code: 128.
fatal: Unable to create '/Users/chriszannoni/Documents/Design/.git/index.lock': File exists.

Another git process seems to be running in this repository, e.g.
an editor opened by 'git commit'. Please make sure all processes
are terminated then try again. If it still fails, a git process
may have crashed in this repository earlier:
remove the file manually to continue.

2017-09-21T16:55:44.703Z - info: [main] Error report submitted
2017-09-21T16:55:54.368Z - info: [ui] Executing getStatus: git status --untracked-files=all --branch --porcelain=2 -z (took 1.396s)
2017-09-21T16:57:06.239Z - info: [ui] Executing exportCommit: git archive ff17f3b04931a2536b682bb2dc16616d8f1a398c -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/ff17f3b04931a2536b682bb2dc16616d8f1a398c.tar (took 1.171s)
2017-09-21T16:57:32.114Z - info: [ui] Executing getStatus: git status --untracked-files=all --branch --porcelain=2 -z (took 3.571s)
2017-09-21T16:59:45.396Z - info: [ui] Executing exportCommit: git archive ff17f3b04931a2536b682bb2dc16616d8f1a398c -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/ff17f3b04931a2536b682bb2dc16616d8f1a398c.tar (took 1.188s)
2017-09-21T16:59:48.855Z - info: [ui] Executing exportCommit: git archive 64e0c24a53bd4b2b31dfb9688be28a5641c4bd7e -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/64e0c24a53bd4b2b31dfb9688be28a5641c4bd7e.tar (took 1.191s)
2017-09-21T17:02:39.377Z - info: [ui] Executing getStatus: git status --untracked-files=all --branch --porcelain=2 -z (took 1.109s)
2017-09-21T17:05:27.321Z - info: [ui] Executing exportCommit: git archive ff17f3b04931a2536b682bb2dc16616d8f1a398c -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/ff17f3b04931a2536b682bb2dc16616d8f1a398c.tar (took 1.724s)
2017-09-21T17:05:27.923Z - info: [ui] Executing exportCommit: git archive ff17f3b04931a2536b682bb2dc16616d8f1a398c -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/ff17f3b04931a2536b682bb2dc16616d8f1a398c.tar (took 1.764s)
2017-09-21T17:05:30.158Z - info: [ui] Executing exportCommit: git archive ff17f3b04931a2536b682bb2dc16616d8f1a398c -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/ff17f3b04931a2536b682bb2dc16616d8f1a398c.tar (took 1.198s)
2017-09-21T17:08:48.062Z - info: [ui] Executing exportCommit: git archive ff17f3b04931a2536b682bb2dc16616d8f1a398c -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/ff17f3b04931a2536b682bb2dc16616d8f1a398c.tar (took 1.228s)
2017-09-21T17:08:50.781Z - info: [ui] Executing exportCommit: git archive ff17f3b04931a2536b682bb2dc16616d8f1a398c -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/ff17f3b04931a2536b682bb2dc16616d8f1a398c.tar (took 1.182s)
2017-09-21T17:09:37.356Z - info: [ui] Executing exportCommit: git archive ff17f3b04931a2536b682bb2dc16616d8f1a398c -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/ff17f3b04931a2536b682bb2dc16616d8f1a398c.tar (took 1.045s)
2017-09-21T17:09:37.927Z - info: [ui] Executing exportCommit: git archive ff17f3b04931a2536b682bb2dc16616d8f1a398c -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/ff17f3b04931a2536b682bb2dc16616d8f1a398c.tar (took 1.409s)
2017-09-21T17:09:39.011Z - info: [ui] Executing exportCommit: git archive ff17f3b04931a2536b682bb2dc16616d8f1a398c -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/ff17f3b04931a2536b682bb2dc16616d8f1a398c.tar (took 1.504s)
2017-09-21T17:09:39.609Z - info: [ui] Executing exportCommit: git archive ff17f3b04931a2536b682bb2dc16616d8f1a398c -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/ff17f3b04931a2536b682bb2dc16616d8f1a398c.tar (took 1.203s)
2017-09-21T17:09:41.274Z - info: [ui] Executing exportCommit: git archive ff17f3b04931a2536b682bb2dc16616d8f1a398c -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/ff17f3b04931a2536b682bb2dc16616d8f1a398c.tar (took 1.152s)
2017-09-21T17:09:44.959Z - info: [ui] Executing exportCommit: git archive ff17f3b04931a2536b682bb2dc16616d8f1a398c -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/ff17f3b04931a2536b682bb2dc16616d8f1a398c.tar (took 1.149s)
2017-09-21T17:09:46.943Z - info: [ui] Executing exportCommit: git archive ff17f3b04931a2536b682bb2dc16616d8f1a398c -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/ff17f3b04931a2536b682bb2dc16616d8f1a398c.tar (took 1.299s)
2017-09-21T17:09:56.030Z - info: [ui] Executing exportCommit: git archive ff17f3b04931a2536b682bb2dc16616d8f1a398c -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/ff17f3b04931a2536b682bb2dc16616d8f1a398c.tar (took 1.183s)
2017-09-21T17:09:56.445Z - info: [ui] Executing exportCommit: git archive ff17f3b04931a2536b682bb2dc16616d8f1a398c -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/ff17f3b04931a2536b682bb2dc16616d8f1a398c.tar (took 1.051s)
2017-09-21T17:09:57.357Z - info: [ui] Executing exportCommit: git archive ff17f3b04931a2536b682bb2dc16616d8f1a398c -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/ff17f3b04931a2536b682bb2dc16616d8f1a398c.tar (took 1.437s)
2017-09-21T17:09:57.726Z - info: [ui] Executing exportCommit: git archive 64e0c24a53bd4b2b31dfb9688be28a5641c4bd7e -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/64e0c24a53bd4b2b31dfb9688be28a5641c4bd7e.tar (took 1.274s)
2017-09-21T17:09:58.248Z - info: [ui] Executing exportCommit: git archive 64e0c24a53bd4b2b31dfb9688be28a5641c4bd7e -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/64e0c24a53bd4b2b31dfb9688be28a5641c4bd7e.tar (took 1.091s)
2017-09-21T17:09:58.769Z - info: [ui] Executing exportCommit: git archive 64e0c24a53bd4b2b31dfb9688be28a5641c4bd7e -o /Users/chriszannoni/Library/Application Support/Kactus/previews/1/64e0c24a53bd4b2b31dfb9688be28a5641c4bd7e.tar (took 1.405s)

**Comment by mathieudutour** [comment]
is it maybe a change to a text with a shared text style? I can't really reproduce.
Anyhow, I don't really have any control over what sketch changes when you change the text of an artboard :/

**Comment by ChrisZannoni** [comment]
So even if I just open the file and hit save without making changes it shows that weird archive value changed. Do you know of any sketch settings that could cause this issue? FYI I am using sketch version 46.2

**Comment by mathieudutour** [comment]
hum that's really weird. do you have any sketch plugins that could interfer?

**Comment by ChrisZannoni** [comment]
So the only plugins I did have, which I have now uninstalled, were the "git" plugin and the "User Flow" plugin. But again I have uninstalled them and still have this issue.

**Comment by mathieudutour** [comment]
could you copy paste the diff here so that I can try to decode it and see what actually changes?

**Comment by ChrisZannoni** [comment]
Example diff for file:
Wearsafe App- iOS 2017/> More/007D8B1D-7B43-4997-B120-75AD63B787B9/71F11319-4F0C-4688-91C0-883D3E1D9082/ACEFCA47-1A6B-4AE8-A2FE-B946244B8D5B.json

Note that the diff shows that the "_archive" value has changed.

**Before:** 

```
{
  "_class": "text",
  "attributedString": {
    "_class": "MSAttributedString",
    "archivedAttributedString": {
      "_archive": "YnBsaXN0MDDUAQIDBAUGbG1YJHZlcnNpb25YJG9iamVjdHNZJGFyY2hpdmVyVCR0b3ASAAGGoK8QGwcIDxAeHyAhIiMnLzAxMjM6PklMT1NbX2NmaFUkbnVsbNMJCgsMDQ5YTlNTdHJpbmdWJGNsYXNzXE5TQXR0cmlidXRlc4ACgBqAA1ZjYW5jZWzTERIKExgdV05TLmtleXNaTlMub2JqZWN0c6QUFRYXgASABYAGgAekGRobHIAIgAmAEYAVgBlWTlNLZXJuXxAfTVNBdHRyaWJ1dGVkU3RyaW5nRm9udEF0dHJpYnV0ZV8QEE5TUGFyYWdyYXBoU3R5bGVXTlNDb2xvchAA0gokJSZfEBpOU0ZvbnREZXNjcmlwdG9yQXR0cmlidXRlc4AQgArTERIKKCsuoikqgAuADKIsLYANgA6AD18QE05TRm9udFNpemVBdHRyaWJ1dGVfEBNOU0ZvbnROYW1lQXR0cmlidXRlI0AsAAAAAAAAW0dvdGhhbS1Cb29r0jQ1NjdaJGNsYXNzbmFtZVgkY2xhc3Nlc18QE05TTXV0YWJsZURpY3Rpb25hcnmjNjg5XE5TRGljdGlvbmFyeVhOU09iamVjdNI0NTs8XxAQTlNGb250RGVzY3JpcHRvcqI9OV8QEE5TRm9udERlc2NyaXB0b3LWP0BBQgpDREVGR0hGW05TQWxpZ25tZW50Wk5TVGFiU3RvcHNfEA9OU01pbkxpbmVIZWlnaHRcTlNUZXh0QmxvY2tzXxAPTlNNYXhMaW5lSGVpZ2h0EAKAACNANAAAAAAAAIASgBTSEgpKS6CAE9I0NU1OV05TQXJyYXmiTTnSNDVQUV8QF05TTXV0YWJsZVBhcmFncmFwaFN0eWxlo1BSOV8QEE5TUGFyYWdyYXBoU3R5bGXUVFVWCldYWVpVTlNSR0JcTlNDb2xvclNwYWNlXxASTlNDdXN0b21Db2xvclNwYWNlRjEgMSAxABABgBaAGNJcCl1eVE5TSUQQAYAX0jQ1YGFcTlNDb2xvclNwYWNlomI5XE5TQ29sb3JTcGFjZdI0NWRlV05TQ29sb3KiZDnSNDU4Z6I4OdI0NWlqXxASTlNBdHRyaWJ1dGVkU3RyaW5noms5XxASTlNBdHRyaWJ1dGVkU3RyaW5nXxAPTlNLZXllZEFyY2hpdmVy0W5vVHJvb3SAAQAIABEAGgAjAC0AMgA3AFUAWwBiAGsAcgB/AIEAgwCFAIwAkwCbAKYAqwCtAK8AsQCzALgAugC8AL4AwADCAMkA6wD+AQYBCAENASoBLAEuATUBOAE6ATwBPwFBAUMBRQFbAXEBegGGAYsBlgGfAbUBuQHGAc8B1AHnAeoB/QIKAhYCIQIzAkACUgJUAlYCXwJhAmMCaAJpAmsCcAJ4AnsCgAKaAp4CsQK6AsACzQLiAukC6wLtAu8C9AL5AvsC/QMCAw8DEgMfAyQDLAMvAzQDNwM8A1EDVANpA3sDfgODAAAAAAAAAgEAAAAAAAAAcAAAAAAAAAAAAAAAAAAAA4U="
    }
  },
  "automaticallyDrawOnUnderlyingPath": false,
  "do_objectID": "ACEFCA47-1A6B-4AE8-A2FE-B946244B8D5B",
  "dontSynchroniseWithSymbol": false,
  "exportOptions": {
    "_class": "exportOptions",
    "exportFormats": [
    ],
    "includedLayerIds": [
    ],
    "layerOptions": 0,
    "shouldTrim": false
  },
  "frame": {
    "_class": "rect",
    "constrainProportions": false,
    "height": 20,
    "width": 45,
    "x": 30,
    "y": 4
  },
  "glyphBounds": "{{0, 4}, {44, 12}}",
  "heightIsClipped": false,
  "isFlippedHorizontal": false,
  "isFlippedVertical": false,
  "isLocked": false,
  "isVisible": true,
  "layerListExpandedType": 0,
  "lineSpacingBehaviour": 2,
  "name": "cancel",
  "nameIsFixed": false,
  "resizingConstraint": 47,
  "resizingType": 0,
  "rotation": 0,
  "shouldBreakMaskChain": false,
  "style": {
    "sharedObjectID": "9E06CEBF-82EE-4CCE-9D5D-A1A932E53F9C"
  },
  "textBehaviour": 0,
  "userInfo": {
    "com.animaapp.stc-sketch-plugin": {
      "kModelPropertiesKey": {
        "isAutoWidth": 1
      }
    }
  }
}

```

**After:**

```
{
  "_class": "text",
  "attributedString": {
    "_class": "MSAttributedString",
    "archivedAttributedString": {
      "_archive": "YnBsaXN0MDDUAQIDBAUGbG1YJHZlcnNpb25YJG9iamVjdHNZJGFyY2hpdmVyVCR0b3ASAAGGoK8QGwcIDxAeHyAhIi0wNjo+RkdISUpOUlNbX2NmaFUkbnVsbNMJCgsMDQ5YTlNTdHJpbmdWJGNsYXNzXE5TQXR0cmlidXRlc4ACgBqAA1ZjYW5jZWzTERIKExgdV05TLmtleXNaTlMub2JqZWN0c6QUFRYXgASABYAGgAekGRobHIAIgAyAFIAVgBlfEBBOU1BhcmFncmFwaFN0eWxlXxAfTVNBdHRyaWJ1dGVkU3RyaW5nRm9udEF0dHJpYnV0ZVZOU0tlcm5XTlNDb2xvctYjJCUmCicoKSorLCpbTlNBbGlnbm1lbnRaTlNUYWJTdG9wc18QD05TTWluTGluZUhlaWdodFxOU1RleHRCbG9ja3NfEA9OU01heExpbmVIZWlnaHQQAoAAI0A0AAAAAAAAgAmAC9ISCi4voIAK0jEyMzRaJGNsYXNzbmFtZVgkY2xhc3Nlc1dOU0FycmF5ojM1WE5TT2JqZWN00jEyNzhfEBdOU011dGFibGVQYXJhZ3JhcGhTdHlsZaM3OTVfEBBOU1BhcmFncmFwaFN0eWxl0go7PD1fEBpOU0ZvbnREZXNjcmlwdG9yQXR0cmlidXRlc4ATgA3TERIKP0JFokBBgA6AD6JDRIAQgBGAEl8QE05TRm9udFNpemVBdHRyaWJ1dGVfEBNOU0ZvbnROYW1lQXR0cmlidXRlI0AsAAAAAAAAW0dvdGhhbS1Cb29r0jEyS0xfEBNOU011dGFibGVEaWN0aW9uYXJ5o0tNNVxOU0RpY3Rpb25hcnnSMTJPUF8QEE5TRm9udERlc2NyaXB0b3KiUTVfEBBOU0ZvbnREZXNjcmlwdG9yEADUVFVWCldYWVpVTlNSR0JcTlNDb2xvclNwYWNlXxASTlNDdXN0b21Db2xvclNwYWNlRjEgMSAxABABgBaAGNJcCl1eVE5TSUQQAYAX0jEyYGFcTlNDb2xvclNwYWNlomI1XE5TQ29sb3JTcGFjZdIxMmRlV05TQ29sb3KiZDXSMTJNZ6JNNdIxMmlqXxASTlNBdHRyaWJ1dGVkU3RyaW5noms1XxASTlNBdHRyaWJ1dGVkU3RyaW5nXxAPTlNLZXllZEFyY2hpdmVy0W5vVHJvb3SAAQAIABEAGgAjAC0AMgA3AFUAWwBiAGsAcgB/AIEAgwCFAIwAkwCbAKYAqwCtAK8AsQCzALgAugC8AL4AwADCANUA9wD+AQYBEwEfASoBPAFJAVsBXQFfAWgBagFsAXEBcgF0AXkBhAGNAZUBmAGhAaYBwAHEAdcB3AH5AfsB/QIEAgcCCQILAg4CEAISAhQCKgJAAkkCVQJaAnACdAKBAoYCmQKcAq8CsQK6AsACzQLiAukC6wLtAu8C9AL5AvsC/QMCAw8DEgMfAyQDLAMvAzQDNwM8A1EDVANpA3sDfgODAAAAAAAAAgEAAAAAAAAAcAAAAAAAAAAAAAAAAAAAA4U="
    }
  },
  "automaticallyDrawOnUnderlyingPath": false,
  "do_objectID": "ACEFCA47-1A6B-4AE8-A2FE-B946244B8D5B",
  "dontSynchroniseWithSymbol": false,
  "exportOptions": {
    "_class": "exportOptions",
    "exportFormats": [
    ],
    "includedLayerIds": [
    ],
    "layerOptions": 0,
    "shouldTrim": false
  },
  "frame": {
    "_class": "rect",
    "constrainProportions": false,
    "height": 20,
    "width": 45,
    "x": 30,
    "y": 4
  },
  "glyphBounds": "{{0, 4}, {44, 12}}",
  "heightIsClipped": false,
  "isFlippedHorizontal": false,
  "isFlippedVertical": false,
  "isLocked": false,
  "isVisible": true,
  "layerListExpandedType": 0,
  "lineSpacingBehaviour": 2,
  "name": "cancel",
  "nameIsFixed": false,
  "resizingConstraint": 47,
  "resizingType": 0,
  "rotation": 0,
  "shouldBreakMaskChain": false,
  "style": {
    "sharedObjectID": "9E06CEBF-82EE-4CCE-9D5D-A1A932E53F9C"
  },
  "textBehaviour": 0,
  "userInfo": {
    "com.animaapp.stc-sketch-plugin": {
      "kModelPropertiesKey": {
        "isAutoWidth": 1
      }
    }
  }
}
```

**Comment by ChrisZannoni** [comment]
FYI once I commit those hundreds of changes, then make another change that second change shows up just fine. However, then when I:

- submit those changes to the feature branch
- close kactus and close sketch
- Move to another branch via terminal
- delete that feature locally. 
- open up kactus and select that remote branch
- Open up the sketch file then hit save it still shows that hundreds of files have changed. 

This also happens when I pull from a remote branch as well.

**Comment by mathieudutour** [comment]
hum I spent 2 days trying to fix this and I believe it's a sketch bug. The weird thing is that it definitely didn't happen with sketch 43 and 44. I'm not sure about 45. 

I reported it and i'll see with them what we can do about that

**Comment by mathieudutour** [comment]
for later, the step to reproduce are simply:
* save
* commit
* close the file
* reopen the file
* save

**Comment by ChrisZannoni** [comment]
One suggestion I have for this is maybe in the kactus ui separate auto generated file changes and user file changes. Because the main problem right now is any deliberate change is lost in all the other autogenerated changes.

**Comment by mathieudutour** [comment]
But there is no way for me to know, as you can see, it's a pretty cryptic string

**Comment by ChrisZannoni** [comment]
Any progress on this? This seems like a big issue as it basically renders your product useless for every single user.

**Comment by mathieudutour** [comment]
Yes I'm working on it on the sketch side, I'll keep you updated

**Comment by azpang** [comment]
Thanks for working on this! I just wanted to bump this problem as well. We are working with a symbol library and when we change 1 symbol, Kactus detects 277 changes. The diff is always the "_archive" string. Our team really hopes we can use this tool to change how we are delivering our designs and would love to collaborate with you.

**Comment by matteodesanti** [comment]
Hi @mathieudutour , 
I work for the Digital Transformation Team of the Italian Government. You can read more about us here https://teamdigitale.governo.it/en/
We are trying to build a UI-Kit for all the institutional Italian digital products. The plan is to do that in a pure open-source way, involving all the Italian UI Designers in this critical challenge. Because of that, we would be very happy to use kactus: it is perfect for our purpose. 
But as soon as we started to experiment kactus, we encountered the bug treated in this issue.
The question is: how can we help to accelerate the bugfix process?  Or shall we report that issue to the sketch development team?
Let me know, please: now we are blocked by this problem. 
Tks!

**Comment by mathieudutour** [comment]
We (the sketch team) are working on a more general fix for it, should be rolled out in sketch 49 or 50.

**Comment by mathieudutour** [comment]
probably not no. but maybe there was a `git status` or `git fetch`  running

**Comment by JaseMiller** [comment]
I think I'm having several issues that are related to this one. Rather than opening a new issue I'll start by describing what I'm seeing here. Mathieu, if you think this might be a different set of issues, let me know and I'll open a new one with screenshot, logs, and details.

* With a handful of changes to a tracked and previously committed Sketch file (admittedly a large one) I see thousands of changes showing up in Kactus. 
* The process displaying those changes is really slow and if I let it do its thing for an hour it typically shows over 20,000 changes to be committed. I am using some symbols, but I only changed a couple of them that should affect at most about 20 artboards in a prototype.
* I'm unable to click the the commit title or description while the changes are adding up. The field is unresponsive most of the time.
* Working with the same repo earlier in the week didn't generate these same issues. Changes were counting in the 100-200's for each commit and the app was responding fine.

Thanks. I still love what we can do with Kactus; just trying to contribute to help it be all it can be! :)

**Comment by culturetim** [comment]
Did this get fixed in the new Sketch 49 release?

**Comment by mathieudutour** [comment]
No it will be fixed in Sketch 50

**Comment by mathieudutour** [comment]
Fixed in Sketch 50 beta :tada:

**Comment by mathieudutour** [comment]
to be clear, it should happen once more when you first save your file with Sketch 50: it will transform the file to the new format.

The new format should look like this for a text layer:
```
  "attributedString": {
    "_class": "attributedString",
    "attributes": [
      {
        "_class": "stringAttribute",
        "attributes": {
          "MSAttributedStringColorAttribute": {
            "_class": "color",
            "alpha": 1,
            "blue": 0,
            "green": 0,
            "red": 0
          },
          "MSAttributedStringFontAttribute": {
            "_class": "fontDescriptor",
            "attributes": {
              "name": "CircularStd-Book",
              "size": 18
            }
          },
          "kerning": 0,
          "paragraphStyle": {
            "_class": "paragraphStyle",
            "alignment": 2,
            "allowsDefaultTighteningForTruncation": 0,
            "baseWritingDirection": 0,
            "tighteningFactorForTruncation": 0.05
          },
          "textStyleVerticalAlignmentKey": 0
        },
        "length": 14,
        "location": 0
      }
    ],
    "string": "Type something"
  },
```
instead of being a `__archived` string

**Comment by hungrysquirrel** [comment]
thanks working much better now. Just had to to that transition commit in my
terminal
